src/Analysis/Engine/Impl/Interpreter/Ast/AstPythonMultipleMembers.cs

```diff
@@ -134,56 +101,102 @@ class AstPythonMultipleMembers : IPythonMultipleMembers, ILocatedMember {
             return default(T);
         }
 
-        public IReadOnlyList<IMember> Members {
-            get {
-                EnsureMembers();
+        #region IMemberContainer
+        public IReadOnlyList<IMember> GetMembers() {
+            lock (_lock) {
+                if (_resolvedMembers != null) {
+                    return _resolvedMembers;
+                }
+
+                var unresolved = _members.OfType<ILazyMember>().ToArray();
+                if (unresolved.Length > 0) {
+                    // Publish non-lazy members immediately. This will prevent recursion
+                    // into EnsureMembers while we are resolving lazy values.
+                    var resolved = _members.Where(m => !(m is ILazyMember)).ToList();
+                    _resolvedMembers = resolved.ToArray();
+
+                    foreach (var lm in unresolved) {
+                        var m = lm.Get();
```

Comment:
It is dangerous to call this under lock. `ILazyMember.Get()` triggers `Module.Imported` which can start module analysis.

Reply:
Yeah but its been like this forever... Suggestions?